front/lib/api/assistant/participants.ts

```diff
@@ -0,0 +1,115 @@
+import type {
+  AgentParticipantType,
+  ConversationParticipantsType,
+  ConversationWithoutContentType,
+  LightAgentConfigurationType,
+  ModelId,
+  Result,
+  UserParticipantType,
+} from "@dust-tt/types";
+import { Err, formatUserFullName, Ok } from "@dust-tt/types";
+import { Op } from "sequelize";
+
+import { getAgentConfiguration } from "@app/lib/api/assistant/configuration";
+import type { Authenticator } from "@app/lib/auth";
+import { AgentMessage, Message, User, UserMessage } from "@app/lib/models";
+
+async function fetchAllUsersById(
+  userIds: ModelId[]
+): Promise<UserParticipantType[]> {
+  const users = (
+    await User.findAll({
+      attributes: ["firstName", "lastName", "imageUrl", "username"],
+      where: {
+        id: {
+          [Op.in]: userIds,
+        },
+      },
+    })
+  ).filter((u) => u !== null) as User[];
+
+  return users.map((u) => ({
+    fullName: formatUserFullName(u),
+    pictureUrl: u.imageUrl,
+    username: u.username,
+  }));
+}
+
+async function fetchAllAgentsById(
+  auth: Authenticator,
+  agentConfigurationIds: string[]
+): Promise<AgentParticipantType[]> {
+  // TODO(2024-3-25 flav) Support fetching many agents by id.
+  const agents = (
+    await Promise.all(
+      agentConfigurationIds.map((agentConfigId) => {
+        return getAgentConfiguration(auth, agentConfigId);
+      })
+    )
+  ).filter((a) => a !== null) as LightAgentConfigurationType[];
+
+  return agents.map((a) => ({
+    configurationId: a.sId,
+    name: a.name,
+    pictureUrl: a.pictureUrl,
+  }));
+}
+
+export async function fetchConversationParticipants(
+  auth: Authenticator,
+  conversation: ConversationWithoutContentType
+): Promise<Result<ConversationParticipantsType, Error>> {
+  const owner = auth.workspace();
+  if (!owner) {
+    return new Err(new Error("Unexpected `auth` without `workspace`."));
+  }
+
+  const messages = await Message.findAll({
+    where: {
+      conversationId: conversation.id,
+    },
+    order: [["rank", "ASC"]],
+    include: [
+      {
+        model: UserMessage,
+        as: "userMessage",
+        required: false,
+        attributes: ["userId"],
+      },
+      {
+        model: AgentMessage,
+        as: "agentMessage",
+        required: false,
+        attributes: ["agentConfigurationId"],
+      },
+    ],
+  });
+
+  const { agentConfigurationIds, userIds } = messages.reduce<{
+    agentConfigurationIds: Set<string>;
+    userIds: Set<ModelId>;
+  }>(
+    (acc, m) => {
+      const { agentMessage, userMessage } = m;
+
+      if (agentMessage) {
+        acc.agentConfigurationIds.add(agentMessage.agentConfigurationId);
+      } else if (userMessage && userMessage.userId) {
+        acc.userIds.add(userMessage.userId);
+      }
+
+      return acc;
+    },
+    { agentConfigurationIds: new Set(), userIds: new Set() }
+  );
+
+  const [users, agents] = await Promise.all([
+    fetchAllUsersById([...userIds]),
+    fetchAllAgentsById(auth, [...agentConfigurationIds]),
```

Comment:
would this be kind of the same as using the `conversationId` [view](https://github.com/dust-tt/dust/blob/ae2c11ce26125abb47ea99ae467ebfd4a3903a35/types/src/front/assistant/agent.ts#L126) ?

Reply:
Yep, will take a look and fix in another PR.